utils/node_local_test.py

```diff
@@ -0,0 +1,798 @@
+#!/usr/bin/python3
+
+"""Test code for dfuse"""
+
+import os
+import sys
+import time
+import uuid
+import json
+import signal
+import subprocess
+import tempfile
+import pickle
+
+from collections import OrderedDict
+
+class DFTestFail(Exception):
+    """Used to indicate test failure"""
+    pass
+
+def umount(path):
+    """Umount dfuse from a given path"""
+    cmd = ['fusermount', '-u', path]
+    ret = subprocess.run(cmd)
+    print('rc from umount {}'.format(ret.returncode))
+    return ret.returncode
+
+def load_conf():
+    """Load the build config file"""
+    file_self = os.path.dirname(os.path.abspath(__file__))
+    json_file = None
+    while True:
+        new_file = os.path.join(file_self, '.build_vars.json')
+        if os.path.exists(new_file):
+            json_file = new_file
+            break
+        file_self = os.path.dirname(file_self)
+        if file_self == '/':
+            raise Exception('build file not found')
+    ofh = open(json_file, 'r')
+    conf = json.load(ofh)
+    ofh.close()
+    return conf
+
+class DaosServer():
+    """Manage a DAOS server instance"""
+
+    def __init__(self, conf):
+        self.running = False
+
+        self._sp = None
+        self.conf = conf
+        self._agent = None
+        self.agent_dir = None
+
+        socket_dir = '/tmp/dnt_sockets'
+        if not os.path.exists(socket_dir):
+            os.mkdir(socket_dir)
+        if os.path.exists('/tmp/dnt_server.log'):
+            os.unlink('/tmp/dnt_server.log')
+
+        self._agent_dir = tempfile.TemporaryDirectory(prefix='dnt_agent_')
+        self.agent_dir = self._agent_dir.name
+
+    def __del__(self):
+        if self.running:
+            self.stop()
+
+    def start(self):
+        """Start a DAOS server"""
+
+        daos_server = os.path.join(self.conf['PREFIX'], 'bin', 'daos_server')
+
+        self_dir = os.path.dirname(os.path.abspath(__file__))
+
+        server_config = os.path.join(self_dir, 'nlt_server.yaml')
+
+        cmd = [daos_server, '--config={}'.format(server_config),
+               'start', '-t' '4', '--insecure', '-d', self.agent_dir,
+               '--recreate-superblocks']
```

Comment:
We should refrain from using "--recreate-superblocks" for testing as per my last conversation with Tom Nabarro.

Reply:
I saw that discussion and thought of this but didn't Tom decide to keep the option?  This is actually local test code I've been using for a while so it may well be that this isn't needed any more, I'll try it.